### ISSUE TITLE
Mikro ORM doesn't allow passing "server" to MSSQL

### ISSUE THREAD

**Issue by SteadEXE** [mikro-orm/mikro-orm#5440]
### Describe the bug

Hello,

It's not possible to pass the "server" parameter to mssql so it is not yet possible to connect to a named instance.
The documentation doesn't mention it either.

Tedious needs "server" option in format "HOST\INSTANCE" in order to allow connecting to a named instance of SQL Server.

### Reproduction

MikroOrmModule.forRoot({
      driver: MsSqlDriver,
      dbName: 'my_awesome_database',
      user: 'sa',
      password: 'sa password',
      server: 'LOCALHOST\\SQLEXPRESS2022' // <-- this is mendatory to be able to login to the local named instance,
      driverOptions: {
        encrypt: false,
      },
      debug: true,
    }),

### What driver are you using?

None

### MikroORM version

6.2.0

### Node.js version

v21.7.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
You can use `driverOptions` here too. I will have a look if we can map such value from host option, I dont see why we would need anything else.

**Comment by SteadEXE** [comment]
Is driverOptions passed to new Connection(/* here */) of tedious?
I can get tedious to connect to my database but cannot get Mikro ORM to it yet. You are probably right, it's a option mapping issue I think

**Comment by SteadEXE** [comment]
I could get it working with this configuration (on tedious alone):

```javascript
const connection = new Connection({
    server: 'name of the server',
    options: {
        database: 'my_awesome_database',
        instanceName: 'SQLEXPRESS2022'
    },
    authentication: {
        type: 'default',
        options: {
            user: 'sa',
            password: 'sa password'
        }
    }
});
```

**Comment by B4nan** [comment]
> Is driverOptions passed to new Connection(/* here */) of tedious?

No, as I just said, it's passed to `knex()`, MikroORM does not directly work with tedious.
